### PR TITLE
[TreeView] Improve plugin signature definition

### PR DESCRIPTION
**Pull request by flaviendelangle**
**Purely internal change, no deep review required**
**No change in the end structure of the signature to keep things simple**

The goal here is to not define all the empty params when you create a new plugin.

The simplest custom plugin goes from:

```tsx
type UseCustomPluginSignature = TreeViewPluginSignature<
  {},
  {},
  {},
  {},
  {},
  never,
  []
>;
```

To

```tsx
type UseCustomPluginSignature = TreeViewPluginSignature<{}>
```

This will allow us to easily add new params like `contextValue`, `slots` or `slotProps` without impacting all the plugins.

## Follow up

- Rename `dependantPlugins` into `dependancies`
- Rename `params` into `props`
- Renamed `defaultizedParams` into `defaultizedProps`

### CODE REVIEW

File: packages/x-tree-view/src/internals/corePlugins/useTreeViewInstanceEvents/useTreeViewInstanceEvents.types.ts [hunk 1]
```diff
@@ -19,13 +19,6 @@ export interface UseTreeViewInstanceEventsInstance {
...
-  {},
-  {},
-  UseTreeViewInstanceEventsInstance,
-  {},
-  {},
-  {},
-  never,
-  []
->;
+export type UseTreeViewInstanceEventsSignature = TreeViewPluginSignature<{
```
Comment (flaviendelangle) [inline packages/x-tree-view/src/internals/corePlugins/useTreeViewInstanceEvents/useTreeViewInstanceEvents.types.ts:22]: This file showcase greatly how better the new approach is

File: packages/x-tree-view/src/internals/models/plugin.ts [hunk 2]
```diff
@@ -29,27 +29,31 @@ type TreeViewResponse<TSignature extends TreeViewAnyPluginSignature> = {
...
-  events: TEvents;
-  contextValue: TContextValue;
-  dependantPlugins: TDependantPlugins;
+  params: T extends { params: {} } ? T['params'] : {};
+  defaultizedParams: T extends { defaultizedParams: {} } ? T['defaultizedParams'] : {};
+  instance: T extends { instance: {} } ? T['instance'] : {};
+  events: T extends { events: {} } ? T['events'] : {};
+  state: T extends { state: {} } ? T['state'] : {};
+  contextValue: T extends { contextValue: {} } ? T['contextValue'] : {};
+  models: T extends { defaultizedParams: {}; modelNames: keyof T['defaultizedParams'] }
```
Comment (michelengelen) [inline packages/x-tree-view/src/internals/models/plugin.ts:49]: gotta love conditional types! ❤️

**Review by michelengelen — APPROVED** [review]
LGTM

**Review by LukasTy — APPROVED** [review]

**Review by noraleonte — APPROVED** [review]
Awesome improvement! 🎉

File: packages/x-tree-view/src/internals/models/plugin.ts (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-tree-view/src/internals/models/plugin.ts:49]: :laughing: Tbh I would love to have a more concise way to do this, it's super verbose for something super simple

**Pull request merged**